src/textures/VideoTexture.js

```diff
@@ -40,8 +28,49 @@ class VideoTexture extends Texture {
 	update() {
 
 		const video = this.image;
+
+		if ( video == null ) {
+
+			if ( this._callbackId !== null && this._callbackSource !== null ) {
+
+				this._callbackSource.cancelVideoFrameCallback( this._callbackId );
+
+			}
+
+			this._callbackSource = null;
+
+			this._callbackId = null;
+
+			return;
+
+		}
+
 		const hasVideoFrameCallback = 'requestVideoFrameCallback' in video;
 
+		if ( hasVideoFrameCallback && video !== this._callbackSource ) {
+
+			if ( this._callbackId !== null && this._callbackSource !== null ) {
+
+				this._callbackSource.cancelVideoFrameCallback( this._callbackId );
+
+			}
+
+			this._callbackSource = video;
+
+			const update = () => {
+
```

Comment:
```suggestion

				if ( this._callbackSource !== video ) return;

```
A simpler alternative to https://github.com/mrdoob/three.js/pull/24758/commits/14697569da22238762f69a8e7222af0f12e74d45.